server/connectors/deploy-connector-jdbc/src/main/scala/com/prisma/deploy/connector/jdbc/database/ColumnMutactionInterpreters.scala

```diff
@@ -150,54 +110,60 @@ case class UpdateColumnInterpreter(builder: JdbcDeployDatabaseMutationBuilder) e
   }
 
   private def updateFromBeforeStateToAfterState(mutaction: UpdateColumn, schemaBeforeMigration: DatabaseSchema): DBIO[_] = {
-    val before               = mutaction.oldField
-    val after                = mutaction.newField
-    val indexMustBeRecreated = before.isRequired != after.isRequired || before.dbName != after.dbName || before.typeIdentifier != after.typeIdentifier
-
-    def updateColumn = builder.updateColumn(
-      mutaction.project,
-      model = mutaction.model,
-      oldColumnName = before.dbName,
-      newColumnName = after.dbName,
-      newIsRequired = after.isRequired,
-      newIsList = after.isList,
-      newTypeIdentifier = after.typeIdentifier
-    )
-
-    def createColumn = builder.createColumn(
-      mutaction.project,
-      tableName = mutaction.model.dbName,
-      columnName = after.dbName,
-      isRequired = after.isRequired,
-      isUnique = after.isUnique,
-      isList = after.isList,
-      typeIdentifier = after.typeIdentifier
-    )
+    val before          = mutaction.oldField
+    val after           = mutaction.newField
+    val typeChanges     = before.typeIdentifier != after.typeIdentifier
+    val requiredChanges = before.isRequired != after.isRequired
+
+    def updateColumn =
+      builder.updateColumn(mutaction.project,
+                           after,
+                           oldTableName = before.model.dbName,
+                           oldColumnName = before.dbName,
+                           oldTypeIdentifier = before.typeIdentifier)
 
     def removeUniqueConstraint = builder.removeIndex(
       mutaction.project,
       tableName = mutaction.model.dbName,
       indexName = schemaBeforeMigration.table_!(mutaction.model.dbName).indexByColumns_!(before.dbName).name
     )
 
-    def addUniqueConstraint = builder.addUniqueConstraint(
-      mutaction.project,
-      tableName = mutaction.model.dbName,
-      columnName = after.dbName,
-      typeIdentifier = after.typeIdentifier
-    )
-
-    def updateColumnActions = (before.isUnique, indexMustBeRecreated, after.isUnique) match {
-      case (true, true, true)  => Vector(removeUniqueConstraint, updateColumn, addUniqueConstraint)
-      case (true, _, false)    => Vector(removeUniqueConstraint, updateColumn)
-      case (true, false, true) => Vector(updateColumn)
-      case (false, _, false)   => Vector(updateColumn)
-      case (false, _, true)    => Vector(updateColumn, addUniqueConstraint)
+    def addUniqueConstraint = builder.addUniqueConstraint(mutaction.project, after)
+
+    def updateColumnActions = (before.isUnique, typeChanges, requiredChanges, after.isUnique) match {
+      // type changes, after unique
+      case (true, true, true, true)   => Vector(updateColumn, addUniqueConstraint)
```

Comment:
The very explicit cases did not help me understand what the logic was. But the comment did. After reading the comment a single pattern `case(_, true, _, true)` is more descriptive.